### PR TITLE
UserDetail: Enhance user detail view

### LINKED ISSUE #281: Missing `Host/Service` notification `states/types`, `TimePeriods` in user datail
## Describe the bug
The `Host/Service` notification filter `states` and `types`, `TimePeriods` etc. are missing in the user details.

### PR DESCRIPTION
**Pull request by sukhwinder33445**
fix #281 #279

### CODE REVIEW

**Review by nilmerg — CHANGES_REQUESTED** [review]
We should enrich the states and types somehow. How about using a `TagList` for both, a localized label and then link to a filtered notification list with the state/type and user filter applied?

**Review by nilmerg — CHANGES_REQUESTED** [review]
Sorry I have to step back from my suggestion to use a TagList. The filters for the notification list resulting from this are waaaaaay too expensive. We cannot provide such expensive filters by default.

So, please make states and types simple comma separated text again. Also:

* Make them localizable. I'd want to see `Problem`, not `problem`. (And don't dare to use `ucfirst` for this :wink: )
* Show a dash again for no value everywhere. That's how we do it everywhere else and it's sufficient

**Comment by sukhwinder33445** [comment]
> waaaaaay too expensive

You mean in terms of performance? Can you please explain why? it's just doing the same thing as we filter something with two options.

**Comment by nilmerg** [comment]
I have 3.5 million notifications in my history. If I use such a filter, I'll sit there waiting for 4+ minutes to get any result. This is acceptable if the filter is my own. But a link in the UI shouldn't lead to this.

**Comment by sukhwinder33445** [comment]
I see, thank you for the explanation.

File: library/Icingadb/Widget/Detail/UserDetail.php [hunk 1]
```diff
@@ -90,4 +95,68 @@ protected function assemble()
...
+                    $localizedTypes[] = t('Problem');
+                    break;
+                case 'ack':
+                    $localizedTypes[] = t('Acknowledgement');
+                    break;
+                case 'recovery':
+                    $localizedTypes[] = t('Recovery');
+                    break;
+                case 'downtime_start':
+                    $localizedTypes[] = t('DowntimeStart');
```
Comment (nilmerg) [inline library/Icingadb/Widget/Detail/UserDetail.php:114]: It's a label/title, so please use general word boundaries/structure. (i.e. spaces)

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — DISMISSED** [review]

**Comment by sukhwinder33445** [comment]
I am working on separating the states for host and services.

**Comment by nilmerg** [comment]
Yep, also don't forget [this](https://github.com/Icinga/icingadb-web/issues/279#issuecomment-905575411). We should remove the mail and pager from list items. The "list" being a "table" isn't good either: #192

**Review by nilmerg — CHANGES_REQUESTED** [review]
* There is no UNKNOWN state for hosts
* You forgot the OK state in `localizeStates`
* `separateStates` cannot get the result of `localizeStates` as input

**Pull request merged**